bentoml/_internal/io_descriptors/numpy.py

```diff
@@ -348,3 +359,98 @@ async def predict(input: NDArray[np.int16]) -> NDArray[Any]:
         inst.sample_input = sample_input
 
         return inst
+
+    async def from_proto(self, request: pb.Request) -> ext.NpNDArray:
+        """
+        Process incoming protobuf request and convert it to ``numpy.ndarray``
+
+        Args:
+            request: Incoming RPC request message.
+            context: grpc.ServicerContext
+
+        Returns:
+            a ``numpy.ndarray`` object. This can then be used
+             inside users defined logics.
+        """
+        from bentoml.grpc.utils.mapping import dtypepb_to_npdtype_map
+        from bentoml.grpc.utils.mapping import fieldpb_to_npdtype_map
+
+        if request.HasField("ndarray"):
+            if request.ndarray.dtype == pb.NDArray.DTYPE_UNSPECIFIED:
+                dtype = None
+            else:
+                try:
+                    dtype = dtypepb_to_npdtype_map()[request.ndarray.dtype]
```

Comment:
This map should probably just live in this IO descriptor file.

Reply:
sg